### DISCUSSION TITLE
NetNamedPipeBinding timeline

### DISCUSSION THREAD

**Discussion by VersatermPublicSafety** [CoreWCF/CoreWCF#817] [General]
Hi!!

I'm currently looking at moving from using WCF to CoreWCF. Our application uses NetNamedPipeBinding  which doesn't seem to be available in CoreWCF at the moment. I did some digging and found some mentions that it should be coming this year (2022). I was just wondering if there was an update/timeline on this?

**Comment by mconnew** [comment]
I'm hoping to start working on it next week.

**Reply by VersatermPublicSafety** [reply]
Thanks for the response!!

I was wondering if you had an idea of the effort involved so we can get a clearer picture of the timeline involved and plan things on our end accordingly.

We are porting one of our applications over to .Net 6 with a target of January 2023. We are currently using WCF for inter process communication with one of our other applications. If you think there might be a fair amount of effort involved and that it might not be available before then, then we'll need to find alternatives.

**Reply by mconnew** [reply]
The plan is to have it done significantly before then. Unless something massive comes up to derail me, it will be this year. It should only take me a couple of weeks of dedicated time (which rarely ever comes up) to complete all the work.

**Reply by aminm-net** [reply]
Hi @mconnew Is there any updates about NetNamedPipeBinding in CoreWCF?
Is it still in your plan? 
Best, Amin

**Reply by mconnew** [reply]
@aminm-net I've got named pipes mostly done for the client side of WCF. I've got a few things I need to fix up, and still need to do testing, but that side is pretty much done. I still need to add Unix Domain Sockets support on the client, but with the way I reworked things on the client, that should be a couple of days work.
On the CoreWCF side, that implementation is going to be relatively quick. I've already written the code to be modular and reusable between NetTcp and NamedPipes (they share about 95% common code base) when I wrote the NetTcp implementation. The work there is going to be mostly moving code around to a common package out of NetTcp and plumbing for using IHostedService for listening on named pipes. The hard work of translating P/Invoke native calls from .NET Framework to using managed classes in .NET was 90% done for the client side. 
I'll try to remember to update here when I start on the CoreWCF side. But basically yes, still on track to be complete by the end of the year.

**Reply by aminm-net** [reply]
@mconnew Many thanks for nice comprehensive update mate! and good luck with your great work.

**Reply by aminm-net** [reply]
@mconnew 
Hi Matt, is there any news on NetNamedPipeBinding release? 
Best, Amin

**Reply by mconnew** [reply]
https://twitter.com/mattconnew/status/1590834183550951424 😄

**Reply by VersatermPublicSafety** [reply]
Hi @mconnew Just wanted to see if you had anymore news for us on this. :)

**Reply by mconnew** [reply]
We're working on releasing a preview of the client. We ended up doing lots of changes so that's why we're going preview. E.g. it will be .net6+ only on the client from here on out (if you need to work on framework too, you will need to multi-target). We're having problems suppressing the tests which is breaking the CI system (it doesn't like Windows only test projects when we run our tests on everything). We're trying to resolve that and then the client packages will be released in preview.  
I didn't get any more progress on the service side as I caught the flu over the winter break. I really recommend getting the flu shot this year, the variant going around right now is brutal.

**Reply by VersatermPublicSafety** [reply]
Thanks for the update Matt!!
Do you have an idea of the timeline for the release of the preview?
